qmail-remote.c

```diff
@@ -342,6 +437,18 @@ int main(int argc, char **argv)
       relayhost[i] = 0;
     }
     if (!stralloc_copys(&host,relayhost)) temp_nomem();
+#ifdef SMTPUTF8
+  } else
+  if (enable_utf8) {
+      char *asciihost = 0;
+      if (!stralloc_0(&host)) temp_nomem();
+      switch (idn2_lookup_u8(host.s,(uint8_t**)&asciihost,IDN2_NFC_INPUT)) {
+        case IDN2_OK:     break;
+        case IDN2_MALLOC: temp_nomem();
+        default:          perm_dns();
+      }
+      if (!stralloc_copys(&idnhost,asciihost)) temp_nomem();
+#endif
```

Comment:
This ifdef is a little extra tricky! It's only really needed around the `idn2_lookup_u8()` logic. That could be moved into `eai_remote.c` (along with an ifdef), wrapped in a descriptive function name, and called from here. Then we wouldn't need an ifdef here.